distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Regex/Pattern_2.enso

```diff
@@ -0,0 +1,180 @@
+                        # If Nothing, there are no named groups
+                        _ : Nothing -> Error.throw (No_Such_Group.Error name)
+                        _ ->
+                            qq = (read_group_map groups name)
+                            case qq of
+                                _ : Nothing -> Nothing
+                                n : Integer -> n
+                case n of
+                    _ : Integer -> n
+                    _ : Nothing -> Error.throw (No_Such_Group.Error name)
+
+    to_text_debug : Text -> Vector Text
+    to_text_debug self input =
+        Match_Iterator.new self input . to_text_debug
+
+## PRIVATE
+
+    Performs the regex match, and iterates through the results. Yields both
+    the matched parts of the string, and the 'filler' parts between them.
+
+    At each step, it yields a Match_Iterator_Value, whivch has either a filler
+    and a match, or just the final filler. A Match_Iterator_Value.Last value is
+    return at the end, and only at the end.
+type Match_Iterator
+    new : Pattern_2 -> Text -> Match_Iterator
+    new pattern input = Match_Iterator.Value pattern input 0
+
+    Value (pattern : Pattern_2) (input : Text) (cursor : Integer)
+
+    next : Match_Iterator_Value
+    next self =
+        regex_result = self.pattern.internal_regex_object.exec self.input self.cursor
+        case regex_result.isMatch of
+            False ->
+                filler_range = Range.new self.cursor (Text_Utils.char_length self.input)
+                filler_span = (Utf_16_Span.Value filler_range self.input).to_grapheme_span
+                Match_Iterator_Value.Last filler_span
+            True ->
+                match_start = regex_result.getStart 0
+                filler_range = Range.new self.cursor match_start
+                filler_span = (Utf_16_Span.Value filler_range self.input).to_grapheme_span
+                match = Match_2.Value self.pattern regex_result self.input
+                next_cursor = match.end 0
+                next_iterator = Match_Iterator.Value self.pattern self.input next_cursor
+                Match_Iterator_Value.Next filler_span match next_iterator
+
+    to_text_debug : Vector Text
+    to_text_debug self =
+        vb = Vector.new_builder
+        go it = case it.next of
+            Match_Iterator_Value.Next filler match next_it ->
+                vb.append ('\"' + filler.text + '\"')
+                vb.append ("/" + (match.span 0).text + "/")
+                go next_it
+            Match_Iterator_Value.Last filler ->
+                vb.append ('\"' + filler.text + '\"')
+        go self
+        vb.to_vector
+
+type Match_Iterator_Value
+  Next (filler : Span) (match : Match_2) (next_iterator : Match_Iterator)
+  Last (filler : Span)
+
+## PRIVATE
+group_map_contains : Any -> Text -> Boolean
+group_map_contains map elem =
+    members = Polyglot.get_members map
+    as_vec = Range.new 0 (members.length) . map i->
+        members . at i
```

Comment:
Won't that work?
```suggestion
    as_vec = Vector.from_polyglot_array members
```

Would be much simpler to read and may be faster.

If that does not work, I'd at least do:
```suggestion
    as_vec = Vector.new members.length members.at
```

Or maybe even better:
```suggestion
    as_vec = Vector.from_polyglot_array (Array_Proxy.from_proxy_object members)
```

Although I think the first suggestion should work and it should be preferred.

Reply:
Yes, the first one worked.